safehttp/mux.go

```diff
@@ -71,20 +71,42 @@ type ServeMux struct {
 	mux *http.ServeMux
 }
 
-func registerHandlers(mux *http.ServeMux, handlers map[string]map[string]handlerConfig) {
-	for pattern, handlersPerMethod := range handlers {
-		handlersPerMethod := handlersPerMethod
-		mux.HandleFunc(pattern, func(w http.ResponseWriter, r *http.Request) {
-			cfg, ok := handlersPerMethod[r.Method]
-			if !ok {
-				http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
-				return
-			}
-			processRequest(cfg, w, r)
-		})
+func registerHandlers(mux *http.ServeMux, handlers map[string]map[string]*handlerConfig) {
+	for pattern, methodToHandlerConfigMap := range handlers {
+		httpHandler := generateHttpHandlerFunc(methodToHandlerConfigMap)
```

Comment:
Please revert this rename: go/go-style/decisions#variable-names.

> Omit types and type-like words from most variable names.

Reply:
Is this comment related to `methodToHandlerConfigMap`? Happy to drop "Map", but when I was refactoring, I had to pause every time I came across `handlersPerMethod` to understand what is going on. What makes things more difficult here is that it is part of `map[string]map[string]handlerConfig` and one needs to know which of these strings refers to what.
If you aren't familiar with http's handler registration (one handler takes care of all methods by default; safehttp avoids that), it isn't obvious why we need a "pattern -> method -> config" map at all.

Would you be okay with `handlerConfigPerMethod` (and prefer it over `methodToHandlerConfig`)?

Reply:
Yup.